thalo-eventstoredb/src/event_store.rs

```diff
@@ -0,0 +1,340 @@
+use std::{fmt::Debug, vec};
+
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use eventstore::{
+    AppendToStreamOptions, Client, EventData, ExpectedRevision, ReadAllOptions, ReadStreamOptions,
+    StreamPosition, ResolvedEvent,
+};
+use futures::TryFutureExt;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use thalo::{
+    aggregate::{Aggregate, TypeId},
+    event::{AggregateEventEnvelope, EventType},
+    event_store::EventStore,
+};
+use uuid::Uuid;
+
+use crate::Error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ESDBEventPayload {
+    created_at: DateTime<Utc>,
+    aggregate_type: String,
+    aggregate_id: String,
+    event_data: serde_json::Value,
+}
+
+impl ESDBEventPayload {
+    pub fn event_envelope<A>(&self, id: usize) -> Result<AggregateEventEnvelope<A>, Error>
```

Comment:
Perhaps this should be named `fn into_event_envelope` and take `self` instead of `&self`.
This would avoid unecessary clones in case the user doesn't need the event payload.